lean4-infoview/src/infoview/interactiveCode.tsx

```diff
@@ -68,18 +69,93 @@ function TypePopupContents({pos, info, redrawTooltip}: {pos: DocumentPosition, i
   } else return <>Loading..</>
 }
 
-/** Tags in code represent values which can be hovered over to display extra info. */
-function InteractiveCodeTag({pos, tag: ct, fmt}: InteractiveTagProps<CodeToken>) {
+/** Tagged spans can be hovered over to display extra info stored in the associated `SubexprInfo`. */
+function InteractiveCodeTag({pos, tag: ct, fmt}: InteractiveTagProps<SubexprInfo>) {
   const mkTooltip = React.useCallback((redrawTooltip: () => void) =>
     <div className="font-code tl pre-wrap">
       <TypePopupContents pos={pos} info={ct.info}
         redrawTooltip={redrawTooltip} />
     </div>, [pos.uri, pos.line, pos.character, ct.info])
+
+  // We mimick the VSCode ctrl-hover and ctrl-click UI for go-to-definition
+  const rs = React.useContext(RpcContext)
+  const ec = React.useContext(EditorContext)
+  const [hoverState, setHoverState] = React.useState<HoverState>('off')
+
+  const [goToLoc, setGoToLoc] = React.useState<Location | undefined>(undefined)
+  const fetchGoToLoc = React.useCallback(async () => {
+    if (goToLoc !== undefined) return goToLoc
+    try {
+      const lnks = await getGoToLocation(rs, pos, 'definition', ct.info)
+      if (lnks !== undefined && lnks.length > 0) {
+        const loc = { uri: lnks[0].targetUri, range: lnks[0].targetSelectionRange }
+        setGoToLoc(loc)
+        return loc
+      }
+    } catch(e) {
+      console.error('Error in go-to-definition: ', JSON.stringify(e))
+    }
+    return undefined
+  }, [rs, pos.uri, pos.line, pos.character, ct.info, goToLoc])
+
+  React.useEffect(() => {
+    const onKeyDown = (e : KeyboardEvent) => {
+      if (e.key === 'Control')
+        setHoverState(st => {
+          const newst = st === 'over' ? 'ctrlOver' : st
+          if (newst === 'ctrlOver') void fetchGoToLoc()
+          return newst
+        })
+    }
+
+    const onKeyUp = (e : KeyboardEvent) => {
+      if (e.key === 'Control')
+        setHoverState(st => st === 'ctrlOver' ? 'over' : st)
+    }
+
+    document.addEventListener('keydown', onKeyDown)
+    document.addEventListener('keyup', onKeyUp)
+    return () => {
+      document.removeEventListener('keydown', onKeyDown)
+      document.removeEventListener('keyup', onKeyUp)
+    }
+  }, [fetchGoToLoc])
+
   return (
-    <WithTooltipOnHover tooltipContent={mkTooltip}>
-      <HighlightOnHoverSpan>
+    <WithTooltipOnHover
+      mkTooltipContent={mkTooltip}
+      onClick={(e, next) => {
+        // On ctrl-click, if location is known, go to it in the editor
+        if (e.ctrlKey) void fetchGoToLoc().then(loc => {
+          if (loc === undefined) return
+          setHoverState(st => st === 'over' ? 'ctrlOver' : st)
+          void ec.revealPosition({ uri: loc.uri, ...loc.range.start })
```

Comment:
`loc.range.start` is an interesting choice, there's some weird things you can do with a mouse drag operation so the mouse up occurs some place else.  Currently it seems to do the right thing, (namely vscode sends no onclick if you select more than one word) but you might be able to think of some ways to break it...

Reply:
I went with just the start of the range since otherwise the whole range is selectec by our `showDocument` implementation. VSCode has some way to show a range by just pulsing rather than selecting it, but it doesn't seem exposed in the API.